### ISSUE TITLE
Install via hacs

### ISSUE THREAD

**Issue by mrueg** [vas3k/home-assistant-berlin-transport#24]
It would be awesome if this integration can be installed via hacs as it would simplify the install as well as make it easier to upgrade the integration. 

Have you considered supporting this? 

https://hacs.xyz/docs/publish/start/

**Comment by phrxmd** [comment]
I support @mrueg's idea. It's not much work, it's really well documented (follow [integrations](https://hacs.xyz/docs/publish/integration/)) for the main integration and [plugins](https://hacs.xyz/docs/publish/plugin) for the JavaScript Lovelace card) and it will make installation and updates much easier for people.

**Comment by vas3k** [comment]
Yes, I agree, it would make life a lot easier for everyone. If anyone has experience and can help me make this happen, I'd be happy to collaborate on a PR from you :)

**Comment by phrxmd** [comment]
Sure, just keep in mind that there are some things that need doing that can‘t be done through a PR and that need to be done by the maintainer, if that‘s fine with you? 

Most importantly, HACS wants integrations and plugins (the Lovelace card) in separate repositories, so you’d need to open a separate repo for the plugin, just take the `.js` file from the `www/` subdirectory here and and put it in the `dist/` subdirectory in the new repo. 

Also it would probably mean that you‘d need to do versioned releases. It‘s not much work and it helps everyone, but it‘s a workflow change so best see if you as the main maintainer are comfortable with it.

**Comment by mrueg** [comment]
So we can now install the repository and the custom component via hacs.

Next step is to configure the integration via config flow: https://developers.home-assistant.io/docs/config_entries_config_flow_handler/

**Comment by mrueg** [comment]
Config flow is here: https://github.com/vas3k/home-assistant-berlin-transport/pull/26

**Comment by vas3k** [comment]
Yeap, I think I got it working via HACS + config flow. Nice!

<img width="379" alt="Screen 2024-01-10 at 10 52 39" src="https://github.com/vas3k/home-assistant-berlin-transport/assets/176344/079e26ef-7f8a-4701-9180-678ddc1d6321">

**Comment by mrueg** [comment]
Great to see it working! 
Does the flow make sense to you? 

One thing that could be improved is adding a flow to update the entity (right now you need to delete it and readd it)

**Comment by vas3k** [comment]
>  Does the flow make sense to you?

The search works, although I didn't realize at first that it was prompting me to select a stop from the list after it. But when you get used to it, it becomes quite straightforward.

And to add multiple stops, I'd have to create multiple integrations, right?

**Comment by mrueg** [comment]
Yes that would be multiple entities (I hope this works with the lovelace card).
btw. If you have time to create a repository for the lovelace card, I can add a PR to do the integration.

**Comment by vas3k** [comment]
I've made the card repo :) https://github.com/vas3k/home-assistant-berlin-transport-card

But I haven't tested it all together yet

**Comment by mrueg** [comment]
It's working for me with the cards and install seems to work as well. Closing this issue then. Thanks for your support and appreciate you starting this integration @vas3k :)
